cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/TestCommand.java

```diff
@@ -331,6 +321,11 @@ public void execute() {
             this.outStream.println("WARNING: Code coverage generation is not supported with Ballerina native test");
         }
 
+        if (!project.buildOptions().nativeImage() && !project.buildOptions().graalVMBuildOptions().isEmpty()) {
```

Comment:
Does this cover the case where user passes only the CLI flag?

Reply:
This is done after we merge both cli-build-options and ballerina-toml-build-options. So...This condition will be evaluated against the combination